### DISCUSSION TITLE
Defining data on a hemisphere using lat-lon data

### DISCUSSION THREAD

**Discussion by JoyMonteiro** [pyvista/pyvista#3355] [Q&A] [answered]
Hello,

I used the issue https://github.com/pyvista/pyvista-support/issues/67 to be able to plot data on a hemisphere, by using my lat-lon data as cell data. When I clip the dataset, I see something that looks expected, with an apparently contiguous surface.

![index1](https://user-images.githubusercontent.com/7300413/191859460-0cf930bf-eafc-4999-b70a-85a88038dddb.png)

but when I try to compute connectivity for this clipped surface, it is clear that the data at the boundary is not connected through

![index](https://user-images.githubusercontent.com/7300413/191859320-ad38e9e0-7eb0-46eb-8fef-884d934df856.png)

As you see in the above image, the orange and grey regions should have been part of the same connected region but are not.
is there some way to define my grid and data differently so that this data will be recognised as part of the same connected region?

TIA,
Joy

**Comment by banesullivan** [comment, marked as answer]
Before I get to actually addressing your question, I want to give a plug for [`geovista`](https://github.com/bjlittle/geovista) -- an extension to PyVista for planetary/global datasets developed by @bjlittle. It's a seriously great package that I have been using a lot lately in conjunction with [`pyvista-xarray`](https://github.com/pyvista/pyvista-xarray). I thought that these data looked global/geospatial and thought you might benefit from GeoVista as well.

Here's an example of something I've been doing lately:

https://user-images.githubusercontent.com/22067021/191333612-011cab35-b4e1-4a28-ac52-654a589a04ec.mov

**Comment by banesullivan** [comment]
> but when I try to compute connectivity for this clipped surface, it is clear that the data at the boundary is not connected through
>
> As you see in the above image, the orange and grey regions should have been part of the same connected region but are not.
is there some way to define my grid and data differently so that this data will be recognised as part of the same connected region?

Yep, this is likely happening along the meridian where the data are wrapping back to close the surface of the sphere.

I suspect that running a simple `.clean()` filter on the surface mesh after it has been projected to a sphere might combine the points along the merdian and connect the surfaces as they are supposed to be.

Ref https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyDataFilters.clean.html?highlight=clean#pyvista.PolyDataFilters.clean

**Comment by JoyMonteiro** [comment]
geovista looks great, and more or less exactly what I need! Thanks.
